lib/recommendation-server.js

```diff
@@ -46,7 +46,8 @@ RecommendationServer.prototype = {
     this.events.unsubscribe('printable-key', this.search);
   },
   _url: function(query) {
-    return `${this.server}/?q=${encodeURIComponent(query)}`;
+    const server = this.prefs.getCharPref('server');
```

Comment:
We should make sure the user hasn't munged the value. One way to do this is to use the `nsIURIFixup` service, which is available on Services.jsm. I _think_ the following should do it:
```js
// nsIURI instance
const uri = Services.uriFixup(server, Services.uriFixup.FIXUP_FLAG_FIX_SCHEME_TYPOS);
const uriString = uri.asciiSpec;
```
Then pass the uriString into the template string.